src/python/grpcio/grpc/_cython/_cygrpc/aio/server.pyx.pxi

```diff
@@ -255,21 +303,112 @@ cdef class AioServer:
         return self._server.add_http2_port(address,
                                           server_credentials._credentials)
 
+    async def _server_main_loop(self,
+                                object server_started):
+        self._server.start()
+        cdef RPCState rpc_state
+        server_started.set_result(True)
+
+        while True:
+            # When shutdown begins, no more new connections.
+            if self._status != AIO_SERVER_STATUS_RUNNING:
+                break
+
+            rpc_state = await _server_call_request_call(
```

Comment:
I have a slightly different view here. I'm explicitly wanting the error to be ignored only if it is in shutdown phase. Otherwise, `_server_main_loop` is the serving coroutine, if it exits unexpectedly, I hope it raise to application layer to notify users.

One improvement I see is that it could use an `grpc.RpcError`. But the semantic will be a bit weird here, since it is not an RPC specific error.

Reply:
But is `_RequestCallError` a temporary error? if so, we should keep running the loop. If not is more a kind of termination error that should stop the server and communicates the user in some way that the server is stoped.

Reply:
Yes, my intent is to make it a termination error.

I added a `_crash_exception` field for the server, and will raise it at `wait_for_termination`. This is my solution to notify users. Do you have better suggestions?

Reply:
just break the loop and logging that a terminal error happened? and if it's needed add what you added for terminating the server.

If this is possible increases readability, having all of the code condensed into the same place.

But again, up to you.

Reply:
For the first suggestion, I can understand the rational of logging it (and I do logged it in the callback). I think my later change is the superset of what you are suggesting? Or did I miss something here?

Can you explain more about what do you think is the ideal way of structure this part? By the way, I hope the callback of the `_serving_task` covers not only `_RequestCallError` but all type of exceptions.

Reply:
Yeps agree that your modification is a superset and covers this and other scenarios, so let's keep it as you have proposed!